packages/dev/core/src/Meshes/Node/Blocks/mappingBlock.ts

```diff
@@ -0,0 +1,143 @@
+    /**
+     * Gets or sets the mapping type used by the block
+     */
+    @editableInPropertyPage("Mapping", PropertyTypeForEdition.List, "ADVANCED", {
+        notifiers: { rebuild: true },
+        options: [
+            { label: "Spherical", value: MappingTypes.Spherical },
+            { label: "Cylindrical", value: MappingTypes.Cylindrical },
+        ],
+    })
+    public mapping = MappingTypes.Spherical;
+
+    /**
+     * Create a new MappingBlock
+     * @param name defines the block name
+     */
+    public constructor(name: string) {
+        super(name);
+
+        this.registerInput("position", NodeGeometryBlockConnectionPointTypes.Vector3);
+        this.registerInput("center", NodeGeometryBlockConnectionPointTypes.Vector3, true, Vector3.Zero());
+        this.registerOutput("uv", NodeGeometryBlockConnectionPointTypes.Vector2);
+    }
+
+    /**
+     * Gets the current class name
+     * @returns the class name
+     */
+    public getClassName() {
+        return "MappingBlock";
+    }
+
+    /**
+     * Gets the position input component
+     */
+    public get position(): NodeGeometryConnectionPoint {
+        return this._inputs[0];
+    }
+
+    /**
+     * Gets the center input component
+     */
+    public get center(): NodeGeometryConnectionPoint {
+        return this._inputs[1];
+    }
+
+    /**
+     * Gets the output component
+     */
+    public get uv(): NodeGeometryConnectionPoint {
+        return this._outputs[0];
+    }
+
+    protected _buildBlock() {
+        if (!this.position.isConnected) {
+            this.uv._storedFunction = null;
+            this.uv._storedValue = null;
+            return;
+        }
+        const tempDirection = Vector3.Zero();
+
+        const func = (state: NodeGeometryBuildState) => {
+            const position = this.position.getConnectedValue(state) as Vector3;
+            const center = this.center.getConnectedValue(state) as Vector3;
+            const uv = Vector2.Zero();
+
+            switch (this.mapping) {
+                case MappingTypes.Spherical: {
+                    position.subtractToRef(center, tempDirection);
+                    const len = tempDirection.length();
+                    if (len > 0) {
+                        uv.x = Math.acos(tempDirection.y / len) / Math.PI;
+                        if (tempDirection.x !== 0 || tempDirection.z !== 0) {
+                            uv.y = Math.atan2(tempDirection.x, tempDirection.z) / (Math.PI * 2);
+                        }
+                    }
+                    break;
+                }
+                case MappingTypes.Cylindrical: {
+                    position.subtractToRef(center, tempDirection);
+                    const len = tempDirection.length();
+                    if (len > 0) {
+                        uv.x = Math.atan2(tempDirection.x / len, tempDirection.z / len) / (Math.PI * 2);
+                        uv.y = (tempDirection.z + 1.0) / 2.0;

```

Comment:
The code assumes the mesh has a Z extent larger than the other two: Shouldn't we try to determine the axis with the largest extent to perform the calculation?

Reply:
I'm not sure. I prefer to document that the mesh as to be on the y axis (gosh, just realized I used z for height)

Reply:
If we have an automatic mode then we can also use it to detect the cylinder direction but for now I'll keep it simple